### PR TITLE
Add component-config documentation with support for outbound connections

### PR DESCRIPTION
**Pull request by Yasithalalanga**
Resolves https://github.com/wso2-enterprise/choreo/issues/26426

### CODE REVIEW

File: en/docs/develop-components/configure-endpoints-body.md [hunk 1]
```diff
@@ -105,6 +105,115 @@ endpoints:
...
+|----------------------|--------------|---------------------------------------------------------------------------------------|
+| **apiVersion**       | Required     | The version of the `component-config.yaml` file defaults to `core.choreo.dev/v1beta1`.|
+| **kind**             | Required     | The resource type of the file defaults to `ComponentConfig`.                          |
+| **spec.inbound**     | Optional     | The list of inbound connection configurations.                                        |
+| **spec.outbound**    | Optional     | The list of outbound connection configurations.                                       |
+
+
+#### Inbound connection configurations (spec.inbound)
+
+The `spec.inbound` section simplifies the configuration of inbound connections, allowing for the inclusion of a list of endpoints. It fully supports the existing endpoints schema structure. Refer to the [endpoints schema in the documentation](#learn-the-endpointsyaml-file) for further details.
```
Comment (Nashaath) [inline en/docs/develop-components/configure-endpoints-body.md:140]: Shall we say:
In the `spec.inbound` configuration section,  you can list endpoints to set up inbound connections. It follows the existing endpoints schema structure. For more information, see the [endpoints schema documentation](#learn-the-endpointsyaml-file).

File: en/docs/develop-components/configure-endpoints-body.md [hunk 2]
```diff
@@ -105,6 +105,115 @@ endpoints:
...
+
+#### Outbound connection configurations (spec.outbound)
+
+Within the `spec.outbound` section, you can define `serviceReferences`, specifying the service references generated in the internal API marketplace when creating a service connection.
```
Comment (Nashaath) [inline en/docs/develop-components/configure-endpoints-body.md:144]: Shall we say:
In the `spec.outbound` section, you can define `serviceReferences` by specifying the service references generated in the internal API marketplace when creating a service connection.

File: en/docs/develop-components/configure-endpoints-body.md (hunk 1 above)
Comment (Yasithalalanga) [reply en/docs/develop-components/configure-endpoints-body.md:140]: Noted

File: en/docs/develop-components/configure-endpoints-body.md (hunk 2 above)
Comment (Yasithalalanga) [reply en/docs/develop-components/configure-endpoints-body.md:144]: Noted

**Review by Nashaath — APPROVED** [review]

**Pull request merged**